python/ingestion/dataset_utils.py

```diff
@@ -596,3 +553,44 @@ def combine_race_ethnicity(
     df = df.drop(columns=[std_col.RACE_COL, std_col.ETH_COL])
 
     return df
+
+
+def generate_time_df_with_cols_and_types(
+    df: pd.DataFrame,
+    time_cols: list[str],
+    table_type: Literal['current', 'historical'],
+    dem_col: Literal['age', 'race', 'race_and_ethnicity', 'sex'],
+):
+    """
+    Accepts a DataFrame along with list of column names for either current or
+    historical data and generates the appropiate BQ types for each column.
+
+    Parameters:
+    - df: The source DataFrame to be processed.
+    - time_cols: A list of column names related to `table_type` that will be
```

Comment:
instead of `time_cols` can we use something more descriptive like `numerical_cols_to_keep`

Reply:
also somewhere, maybe this doc string, it'd be helpful to specify the types of columns that the user should send in for each time type, e.g. 100k or pct_rate are in both, pct_share are in current and only in historical if there are unknown demographic rows; pct_rel_inequity is only in the historical tables, raw / estimated total are only in the current. maybe mention which frontend tables require which metrics